### PR TITLE
Additional Logging in EventStoreSubscriber

### PR DESCRIPTION
**Pull request by robings**
# Additional Logging in EventStoreSubscriber

## What's Changed?
- Addition of information logging in `EventStoreSubscriber`
- Allows alternative metadata key to be passed to `EventStoreAggregateDispatcher`

### CODE REVIEW

File: src/eventstore/EventStoreSubscriber.cs [hunk 1]
```diff
@@ -452,6 +453,13 @@ private void ProcessEvent(ResolvedEvent resolvedEvent)
                 var checkpointNumber = GetLastProcessedPosition(resolvedEvent);
 
                 WriteCheckpoint(checkpointNumber);
+                _logger.LogInformation(
+                    "Event dispatched from Eventstore subscriber ({0}/{1})",
+                    resolvedEvent.Event.EventStreamId,
+                    resolvedEvent.Event.EventNumber);
+#if DEBUG
+                _logger.LogTrace("Event dispatched {ResolvedEvent}]", resolvedEvent);
```
Comment (rcknight) [inline src/eventstore/EventStoreSubscriber.cs:461]: I think including the actual event content is overkill - if you have event number / stream it is easy to go look up in the event store UI if needed.

**Review by rcknight — CHANGES_REQUESTED** [review]

File: src/eventstore/EventStoreSubscriber.cs [hunk 2]
```diff
@@ -452,6 +453,13 @@ private void ProcessEvent(ResolvedEvent resolvedEvent)
                 var checkpointNumber = GetLastProcessedPosition(resolvedEvent);
 
                 WriteCheckpoint(checkpointNumber);
+                _logger.LogInformation(
```
[the remark this answers is not in the document]
Comment (robings) [reply src/eventstore/EventStoreSubscriber.cs:456]: Changed to trace.

File: src/eventstore/EventStoreSubscriber.cs (hunk 1 above)
Comment (robings) [reply src/eventstore/EventStoreSubscriber.cs:461]: Removed logging of event content.

File: src/eventstore/EventStoreSubscriber.cs [hunk 3]
```diff
@@ -486,6 +497,7 @@ private void WriteCheckpoint(ulong checkpointNumber)
             }
 
             _checkpoint.Write((long)checkpointNumber);
+            _logger.LogInformation("Checkpoint written. Checkpoint number {CheckpointNumber}", checkpointNumber);
```
[the remark this answers is not in the document]
Comment (robings) [reply src/eventstore/EventStoreSubscriber.cs:500]: Changed to trace.

**Review by rcknight — APPROVED** [review]

**Pull request merged**